Mathlib/Topology/Algebra/Category/ProfiniteGrp/Limits.lean

```diff
@@ -0,0 +1,140 @@
+* `QuotientOpenNormalSubgroup` : The functor that maps open normal subgroup of `P` to
+  the quotient group of it (which is finite, as shown by previous lemmas).
+
+* `CanonicalQuotientMap` : The continuous homomorphism from `P` to the limit of the quotient group
+  of open normal subgroup ordered by inclusion of the open normal subgroup.
+
+* `canonicalQuotientMap_surjective` : The `CanonicalQuotientMap` is surjective, proven by
+  demonstrating that its range is dense and closed.
+
+* `OpenNormalSubgroupSubClopenNhdsOfOne` : For any open neighborhood of `1` there is an
+  open normal subgroup contained within it.
+
+* `canonicalQuotientMap_injective` : The `CanonicalQuotientMap` is injective. This is proven by
+  showing that for any element not equal to one, the image of it on the coordinate of
+  the open normal subgroup that doesn't contain it is not equal to 1, thus not in the kernel.
+
+* `ContinuousMulEquivLimitQuotientOpenNormalSubgroup` : The `CanonicalQuotientMap` can serve as a
+  `ContinuousMulEquiv`, with the continuity of other side given by
+  `Continuous.homeoOfEquivCompactToT2`.
+
+-/
+
+universe u
+
+open CategoryTheory Topology TopologicalGroup
+
+namespace ProfiniteGrp
+
+theorem exist_openNormalSubgroup_sub_open_nhd_of_one {G : Type*} [Group G] [TopologicalSpace G]
+    [TopologicalGroup G] [CompactSpace G] [TotallyDisconnectedSpace G] {U : Set G}
+    (UOpen : IsOpen U) (einU : 1 ∈ U) : ∃ H : OpenNormalSubgroup G, (H : Set G) ⊆ U := by
+  rcases ((Filter.HasBasis.mem_iff' ((nhds_basis_clopen (1 : G))) U ).mp <|
+    mem_nhds_iff.mpr (by use U)) with ⟨W, hW, h⟩
+  rcases exist_openNormalSubgroup_sub_clopen_nhd_of_one hW.2 hW.1 with ⟨H, hH⟩
+  exact ⟨H, fun _ a ↦ h (hH a)⟩
+
+section
+
+instance (P : ProfiniteGrp) : SmallCategory (OpenNormalSubgroup P) :=
+  Preorder.smallCategory (OpenNormalSubgroup ↑P.toProfinite.toTop)
+
+/-- Define the functor from OpenNormalSubgroup of a profinite group to the quotient group of it as
+  a `FiniteGrp` -/
+def QuotientOpenNormalSubgroup (P : ProfiniteGrp) :
+    OpenNormalSubgroup P ⥤ FiniteGrp := {
+    obj := fun H => FiniteGrp.of (P ⧸ H.toSubgroup)
+    map := fun {H K} fHK => QuotientGroup.map H.toSubgroup K.toSubgroup (.id _) <|
+        Subgroup.comap_id (N := P) K ▸ leOfHom fHK
+    map_id := fun H => by
+      simp only [QuotientGroup.map_id]
+      rfl
+    map_comp := fun {X Y Z} f g => (QuotientGroup.map_comp_map
+      X.toSubgroup Y.toSubgroup Z.toSubgroup (.id _) (.id _)
+      (Subgroup.comap_id Y.toSubgroup ▸ leOfHom f)
+      (Subgroup.comap_id Z.toSubgroup ▸ leOfHom g)).symm }
+
+/-- Defining the canonical projection from a profinite group to the limit of the quotient groups
+as a subgroup of the product space -/
+def CanonicalQuotientMap (P : ProfiniteGrp.{u}) : P ⟶
+    limit ((QuotientOpenNormalSubgroup P) ⋙ (forget₂ FiniteGrp ProfiniteGrp)) where
+  toFun := fun p => { val := fun H => QuotientGroup.mk p
+                      property := fun _ => rfl }
```

Comment:
```suggestion
  toFun p := { val H := QuotientGroup.mk p
               property _ := rfl }
```
This is the preferred style and there are several other places in this PR where this kind of change can be made.